internal/common/conversion/schema_generation.go

```diff
@@ -11,14 +11,44 @@ import (
 func DataSourceSchemaFromResource(rs schema.Schema, requiredFields []string, overridenFields map[string]dsschema.Attribute) dsschema.Schema {
 	blocks := convertBlocks(rs.Blocks, requiredFields)
 	attrs := convertAttrs(rs.Attributes, requiredFields)
-	for name, attr := range overridenFields {
-		if attr == nil {
-			delete(attrs, name)
-		} else {
-			attrs[name] = attr
+	overrideFields(attrs, overridenFields)
+	ds := dsschema.Schema{Attributes: attrs, Blocks: blocks}
+	UpdateSchemaDescription(&ds)
+	return ds
+}
+
+func PluralDataSourceSchemaFromResource(rs schema.Schema, requiredFields []string, overridenFields, overridenRootFields map[string]dsschema.Attribute, overrideResultsDoc string, hasLegacyFields bool) dsschema.Schema {
+	blocks := convertBlocks(rs.Blocks, nil)
+	if len(blocks) > 0 {
+		panic("blocks not supported yet in auto-generated plural data source schema as they can't go in ListNestedAttribute")
+	}
+	attrs := convertAttrs(rs.Attributes, nil)
+	overrideFields(attrs, overridenFields)
+	rootAttrs := convertAttrs(rs.Attributes, requiredFields)
+	for name := range rootAttrs {
+		if !slices.Contains(requiredFields, name) {
+			delete(rootAttrs, name)
 		}
 	}
-	ds := dsschema.Schema{Attributes: attrs, Blocks: blocks}
+	overrideFields(rootAttrs, overridenRootFields)
+	resultsDoc := "List of returned documents that MongoDB Cloud provides when completing this request."
+	if overrideResultsDoc != "" {
+		resultsDoc = overrideResultsDoc
+	}
+	rootAttrs["results"] = dsschema.ListNestedAttribute{
+		Computed: true,
+		NestedObject: dsschema.NestedAttributeObject{
+			Attributes: attrs,
+		},
+		MarkdownDescription: resultsDoc,
```

Comment:
I guess there is no way we can use a better description here?
Or generic is ok?

Reply:
do you mean to improve the var name resultsDoc, or the text "List of documents that MongoDB ..." ?

Reply:
"List of documents that MongoDB"

Reply:
this is the one we're using right now in most of the resources, how you have a suggestion?